conda_content_trust/common.py

```diff
@@ -181,145 +179,34 @@ class MixinKey:
     PrivateKey and PublicKey classes, specifically.  It provides some
     convenience functions.
     """
+    @classmethod
+    def to_hex(cls, key):
```

Comment:
Why are we converting these into `classmethods`? I don't see a good reason to stop allowing:

```
private.to_hex()
public.to_hex()
```

Reply:
This has something to do with how the code is wrapping cryptography's API, when it's instantiating objects via their `from_private_bytes` and `from_public_bytes` class methods, which we wrap with more generic versions called `from_bytes` for usability (?) reasons. What we get back is their rust_openssl objects, which don't know about the `to_hex` method.

I'm trying to walk back the API design a little since I feel there is way too much magic going on for no perceivable gain. If you have suggestions to resolve this API design, I'm very interested (had not much time to think about it).

Reply:
In the previous version it was using subclass hacks to inject the methods via a mixin, but when cryptography recently switched to the Rust implementation, the hack broke, basically.